From 62463f8f20aad2869bc047fbf526e42f601c354d Mon Sep 17 00:00:00 2001
From: Jiaxing Shi <jiaxshi@qti.qualcomm.com>
Date: Thu, 22 Jan 2026 19:33:00 +0800
Subject: [PATCH] build: remove boost system

Since Boost 1.69, boost_sgiystem is header-only and doesn't need to be
linked. Remove it from find_package and all target_link_libraries calls.

Upstream-Status: Pending

Signed-off-by: Jiaxing Shi <jiaxshi@qti.qualcomm.com>
---
 CMakeLists.txt               | 2 +-
 CMakeModules/OMPLUtils.cmake | 1 -
 demos/CMakeLists.txt         | 1 -
 src/ompl/CMakeLists.txt      | 1 -
 4 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb7797b..bc65550 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ set_package_properties(Boost PROPERTIES
     URL "https://www.boost.org"
     PURPOSE "Used throughout OMPL for data serialization, graphs, etc.")
 set(Boost_USE_MULTITHREADED ON)
-find_package(Boost 1.68 REQUIRED COMPONENTS serialization filesystem system program_options)
+find_package(Boost 1.68 REQUIRED COMPONENTS serialization filesystem program_options)
 
 # on macOS we need to check whether to use libc++ or libstdc++ with clang++
 if(CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
diff --git a/CMakeModules/OMPLUtils.cmake b/CMakeModules/OMPLUtils.cmake
index ddd6f9a..9a63df7 100644
--- a/CMakeModules/OMPLUtils.cmake
+++ b/CMakeModules/OMPLUtils.cmake
@@ -5,7 +5,6 @@ macro(add_ompl_test test_name)
     Boost::program_options
     Boost::serialization
     Boost::filesystem
-    Boost::system
     Boost::unit_test_framework)
   add_test(NAME ${test_name} COMMAND $<TARGET_FILE:${test_name}>)
 endmacro(add_ompl_test)
diff --git a/demos/CMakeLists.txt b/demos/CMakeLists.txt
index 0de3222..4754727 100644
--- a/demos/CMakeLists.txt
+++ b/demos/CMakeLists.txt
@@ -12,7 +12,6 @@ if (OMPL_BUILD_DEMOS)
             ompl::ompl
             Eigen3::Eigen
             Boost::filesystem
-            Boost::system
             Boost::program_options)
     endmacro(add_ompl_demo)
 
diff --git a/omplConfig.cmake.in b/omplConfig.cmake.in
index f1d4785..5f40fe1 100644
--- a/omplConfig.cmake.in
+++ b/omplConfig.cmake.in
@@ -12,7 +12,7 @@ set_and_check(OMPL_INCLUDE_DIRS @PACKAGE_INCLUDE_INSTALL_DIR@)
 
 include ("${CMAKE_CURRENT_LIST_DIR}/omplExport.cmake" )
 include(CMakeFindDependencyMacro)
-set(_@PROJECT_NAME@_boost_components serialization filesystem system)
+set(_@PROJECT_NAME@_boost_components serialization filesystem)
 find_dependency(Boost REQUIRED COMPONENTS ${_@PROJECT_NAME@_boost_components})
 if(Boost_FOUND)
     foreach(_comp ${_@PROJECT_NAME@_boost_components})
diff --git a/src/ompl/CMakeLists.txt b/src/ompl/CMakeLists.txt
index 463930c..82911ef 100644
--- a/src/ompl/CMakeLists.txt
+++ b/src/ompl/CMakeLists.txt
@@ -44,7 +44,6 @@ target_link_libraries(ompl
     PUBLIC
         Boost::filesystem
         Boost::serialization
-        Boost::system
         Eigen3::Eigen
         "$<$<BOOL:${Threads_FOUND}>:Threads::Threads>"
         "$<$<BOOL:${OMPL_HAVE_FLANN}>:flann::flann>"
-- 
2.34.1

